proposals/0160-static-instruction-limit.md

```diff
@@ -0,0 +1,66 @@
+---
+simd: '0160'
+title: Static Instruction Limit
+authors:
+  - Andrew Fitzgerald (anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-07-24
+feature:
+supersedes:
+superseded-by:
+extends:
+---
+
+## Summary
+
+Transactions will fail execution in SVM if the transactions use more than 64
+instructions, including CPI calls.
+This proposal is to make transactions that use more than 64 top-level
+instructions fail sanitization checks and be rejected by the network.
+
+## Motivation
+
+The current limit of 64 instructions is a runtime failure, and makes sense
+in the context of CPIs.
+It is bad user experience to allow creation and submission of transactions that
+have no chance of being executed successfully.
+Several other checks worst-case performance scales with the number of
+top-level instructions in a transaction, so limiting this number can help the
+worst-case performance.
+
+## Alternatives Considered
+
+Do nothing.
```

Comment:
Why not allow them to be included but limit processing to fee payer validation? So before we load the program cache, we check the instruction count and fail the transaction immediately if it exceeds the limit but still allow fee collection?

Reply:
I view it similarly to having the number of signatures not match the required signatures field in the header; its' a basic sanitization check that can be checked extremely cheaply.

checking it does not scale with any aspects of the transaction such as number of instructions, number of account lookup tables, etc.

Reply:
So long as we do it really early then I'm fine with prohibiting such transactions from being included. Can you write up a brief summary of this comment thread in the alternatives section?

Reply:
4398665f2ae72df927fd532effcb6b145903700a